components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -1554,6 +1555,48 @@ func analyseConfigChanges(ctx context.Context, wscfg *Config, w analytics.Writer
 	}
 }
 
+func trackReadiness(ctx context.Context, gitpodService *gitpod.APIoverJSONRPC, cfg *Config, cstate *InMemoryContentState, ideReady *ideReadyState, desktopIdeReady *ideReadyState) {
+	type SupervisorReadiness struct {
+		Kind                string `json:"kind,omitempty"`
+		WorkspaceId         string `json:"workspaceId,omitempty"`
+		WorkspaceInstanceId string `json:"instanceId,omitempty"`
+		Timestamp           int64  `json:"timestamp,omitempty"`
+	}
+	trackFn := func(ctx context.Context, gitpodService *gitpod.APIoverJSONRPC, cfg *Config, kind string) {
+		gitpodService.TrackEvent(ctx, &gitpod.RemoteTrackMessage{
+			Event: "supervisor_readiness",
+			Properties: SupervisorReadiness{
+				Kind:                kind,
+				WorkspaceId:         cfg.WorkspaceID,
+				WorkspaceInstanceId: cfg.WorkspaceInstanceID,
+				Timestamp:           time.Now().Unix(),
```

Comment:
Let's use milliseconds to have a more accurate timestamp
```suggestion
				Timestamp:           time.Now().UnixMilli(),
```

Reply:
I copied this `Timestamp` from somewhere inside Gitpod repo () 🤣